apps/passwd.c

```diff
@@ -306,9 +306,9 @@ static char *md5crypt(const char *passwd, const char *magic, const char *salt)
     out_buf[0] = '$';
     out_buf[1] = 0;
     assert(strlen(magic) <= 4); /* "1" or "apr1" */
-    strncat(out_buf, magic, 4);
-    strncat(out_buf, "$", 1);
-    strncat(out_buf, salt, 8);
+    strncat(out_buf, magic, sizeof(out_buf) - strlen(out_buf) - 1);
+    strncat(out_buf, "$", sizeof(out_buf) - strlen(out_buf) - 1);
+    strncat(out_buf, salt, sizeof(out_buf) - strlen(out_buf) - 1);
```

Comment:
I suggest using `BUF_strlcat` instead.

Reply:
Okay. I did not know this BUF_strlcat...